management/models.py

```diff
@@ -25,6 +32,15 @@ class Cohort(BaseModel):
     group = models.CharField(max_length=50)
     active = models.BooleanField(default=True)
 
+    def delete(self, *args, **kwargs):
+        user = kwargs.get('user')
+        if user is None:
+            user = getattr(kwargs.get('request'), 'user', None)
+
+        create_event(EventNames.CohortDeleted, user=user,
```

Comment:
This probably can just pass the request in since the create_event pulls the user out of the request. (Don't need the user bit)

However you have to probably have to pass the request as a parameter to delete from wherever it's called from. That may lead to an error when you call the super method. To save you some time when you get close the trick is going to be:

`request = kwargs.pop("request", None)`

Reply:
Walked through the changes with Matt. We are now passing the parameters to the dB. Going to add the changes to this commit.